src/Service/WorkloadReportService.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace App\Service;
+
+use App\Model\Reports\WorkloadReportData;
+use App\Model\Reports\WorkloadReportWorker;
+use App\Repository\WorkerRepository;
+use App\Repository\WorklogRepository;
+
+class WorkloadReportService
+{
+    public function __construct(
+        private readonly WorkerRepository $workerRepository,
+        private readonly WorklogRepository $worklogRepository,
+        private readonly DateTimeHelper $dateTimeHelper,
+    ) {
+    }
+
+    /**
+     * Get the workload report based on the specified view mode.
+     *
+     * @param string $viewMode The view mode (default: 'week')
+     *
+     * @return WorkloadReportData The workload report data
+     *
+     * @throws \Exception When an unexpected value for viewMode is provided
+     */
+    public function getWorkloadReport(string $viewMode = 'week'): WorkloadReportData
```

Comment:
viewMode should be changed to an enum: https://www.php.net/manual/en/language.types.enumerations.php

Reply:
The next PR in line (adds to the workload report functionality), renames the current $viewMode and adds an additional option. To avoid conflicts, i will implements this in the next PR for both the options currently using strings.

https://github.com/itk-dev/economics/pull/129